File: CODE/PYTHON/LIBRARY/writing.py

# -- IMPORTS

from pathlib import Path;

from .building import build_gson_text;

# -- FUNCTIONS


def write_file_text(
    file_path,
    file_text
    ):

    Path( file_path ).write_text( file_text, encoding = "utf8" );


# ~~


def write_gson_value(
    file_path,
    value,
    write_file_text_function = write_file_text
    ):

    write_file_text_function( file_path, build_gson_text( value ) );

File: CODE/TYPESCRIPT/FILE/writing.ts
// -- IMPORTS

import { buildGsonText } from "senselogic-gson";
import type { JsonValue } from "./types.ts";

// -- FUNCTIONS

export function writeFileText(
    filePath: string,
    fileText: string
    )
    : void
{
    Deno.writeTextFileSync( filePath, fileText );
}

// ~~

export function writeGsonValue(
    filePath: string,
    value: JsonValue
    )
    : void
{
    writeFileText( filePath, buildGsonText( value ) );
}
